core/src/server/rpc_module.rs

```diff
@@ -738,6 +748,78 @@ impl<Context: Send + Sync + 'static> RpcModule<Context> {
 	}
 }
 
+/// Represent a pending subscription which waits being accepted or rejected.
+///
+/// Warning: you need to call either `PendingSubscription::accept` or `PendingSubscription::reject` otherwise
+/// the subscription will be dropped with an `InvalidParams` error.
+#[derive(Debug)]
+struct InnerPendingSubscription {
+	/// Sink.
+	sink: MethodSink,
+	/// Get notified when subscribers leave so we can exit
+	close_notify: Option<Arc<Notify>>,
+	/// MethodCallback.
+	method: &'static str,
+	/// Unique subscription.
+	uniq_sub: SubscriptionKey,
+	/// Shared Mutex of subscriptions
+	subscribers: Subscribers,
+	/// Request ID.
+	id: Id<'static>,
+}
+
+/// Represent a pending subscription which waits for being accepted or rejected.
+///
+/// This type implement drop for ease of use, such as dropped in error short circuiting via `map_err()?`.
+#[derive(Debug)]
+pub struct PendingSubscription(Option<InnerPendingSubscription>);
```

Comment:
Perhaps we don't need an `Option<InnerPendingSubscription>` now, since we can only reject or accept it once, and each function takes ownership?

Reply:
it won't work with the `drop impl` any better suggestions?

Reply:
Nope, that makes sense! I realised wh ythe Option was here when I got to looking at that drop impl :)